packages/editor/src/store/actions.js

```diff
@@ -474,11 +478,24 @@ export function* savePost( options = {} ) {
 		'getCurrentPost'
 	);
 
-	const editedPostContent = yield select(
+	let editedPostContent = yield select(
 		STORE_KEY,
 		'getEditedPostContent'
 	);
 
+	const template = ( yield select( STORE_KEY, 'getEditorSettings' ) ).template;
```

Comment:
Oh, so you're using the current `template` editor setting as the actual "FSE template". The `template` property is meant to be the `post_content` template while the actual FSE template is something that get stored separately. There's a question to be asked about "naming confusion" and also whether we may want to absorb one use-case in another in the future but I think for now we may want to consider them as two separate entities.

Basically what I had in mind is something like that:

 - a separate `template` prop in the  `EditorProvider` for the FSE template
 - The same way we have `currentPost` and `edits` for the post in the reducer, I think we'd need `currentTemplate` and `templateEdits` maybe.
 - a boolean prop or something like that in the `EditorProvider` to switch between FSE mode (showing the template blocks) and current mode (showing a PostTitle input and the PostContent blocks as root level)
 - Both `core/post-content`  and `core/post-title` blocks should be dynamic blocks that render the actual post title and post content in the frontend.
 - Basically `post-content.php` file shouldn't have to be touched to be loaded in an editor that has a template as well.

Does that make sense?

Reply:
Yeah, that was what I was planning, except I wasn't sure if I should change the way the template was being injected and was going to leave that for last so we could discuss it further.

From my zoom prototyping:

>I started by trying to support insertions, replacements, and removals, but I quickly realized that it would take a pretty widespread refactor due to a lot of assumptions we make in the logic for getting insertion indexes and other data. I don't think the need for this feature warrants those changes. This makes me think that we might have to try a different approach for #16485 where the editor just switches between editing the template and editing post content with a static post title component like it does now.

So for next steps here I see:

- [ ] Give the template its own reducer and make all the necessary refactors. (Use the `'block_editor_settings'` filter to send it to the client in its own, new key.)
- [ ] Implement the "zoom" as a toggle between setting up the editor with the template or post.
- [ ] Implement the render callbacks of the post title and post content blocks.